apps/storefront/src/utils/b3CheckPermissions/base.ts

```diff
@@ -107,38 +109,22 @@ export const getCorrespondsConfigurationPermission = (
   permissions: PermissionsCodesProp[],
   selectCompanyHierarchyId: number,
 ) => {
-  const keys = Object.keys(b2bPermissionsMap);
-
-  const newB3PermissionsList: Record<string, string> = b2bPermissionsMap;
-
-  return keys.reduce((acc, cur: string) => {
-    const param = {
-      code: newB3PermissionsList[cur],
-    };
-
-    const item = checkPermissionCode(param, 'every', permissions || []);
-
-    if (pdpButtonAndOthersPermission.includes(cur)) {
-      const isPdpButtonAndOthersPermission = validateBasePermissionWithComparisonType({
-        code: newB3PermissionsList[cur],
+  const level = selectCompanyHierarchyId
+    ? permissionLevels.COMPANY_SUBSIDIARIES
+    : permissionLevels.USER;
+
+  return mapValues(b2bPermissionsMap, (code, key) => {
+    if (pdpButtonAndOthersPermission.includes(key)) {
+      return validateBasePermissionWithComparisonType({
+        code,
         containOrEqual: 'contain',
-        level: selectCompanyHierarchyId
-          ? permissionLevels.COMPANY_SUBSIDIARIES
-          : permissionLevels.USER,
+        level,
         permissions,
       });
-
-      return {
-        ...acc,
-        [cur]: isPdpButtonAndOthersPermission,
-      };
     }
 
-    return {
-      ...acc,
-      [cur]: item,
-    };
-  }, {} as B2BPermissionsMapParams);
+    return checkPermissionCode({ code }, 'every', permissions || []);
```

Comment:
`permissions || []` - if we provide an empty array, will this ever return true?

I'm guessing if there's nothing to check against, there's no permission to grant?